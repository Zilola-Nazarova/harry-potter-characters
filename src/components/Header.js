import { useState, useRef } from 'react';
import { NavLink } from 'react-router-dom';
import { IoIosArrowBack } from 'react-icons/io';
import styles from '../styles/Header.module.css';
import useOnClickOutside from './useOnClickOutside';

const Header = () => {
  const [dropdown, setDropdown] = useState(false);
  const ref = useRef();
  useOnClickOutside(ref, dropdown, () => setDropdown(false));
  return (
    <header className={styles.header}>
      <nav className={styles.navigation}>
        <ul>
          <li className={styles.arrow}>
            <NavLink to="/">
              <IoIosArrowBack />
            </NavLink>
          </li>
          <li className={styles.title}>
            <NavLink to="/">
              <h1>
                mountains forecast
              </h1>
            </NavLink>
          </li>
          <li>
            <NavLink data-testid="clickAbout" to="/about">
              about
            </NavLink>
          </li>
          <li ref={ref}>
            <button
              type="button"
              onClick={() => setDropdown((prev) => !prev)}
            >
              models
            </button>
            {dropdown && (
              <ul>
                <li>
                  <NavLink to="/models/icon">
                    ICON
                  </NavLink>
                </li>
                <li>
                  <NavLink to="/models/gfs">
                    GFS
                  </NavLink>
                </li>
              </ul>
            )}
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
