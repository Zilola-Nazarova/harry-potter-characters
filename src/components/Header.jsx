import styles from '@/styles/Header.module.css'
import { useState, useRef } from "react";
import { useOnClickOutside } from "./useOnClickOutside";
import { Link } from 'react-router-dom';
import {IoIosArrowBack} from 'react-icons/io';

const Header = () => {
  const [dropdown, setDropdown] = useState(false);
  const ref = useRef();
  useOnClickOutside(ref, dropdown, () => setDropdown(false));
  
  return (
    <header className={styles.header}>
      <nav className={styles.navigation}>
        <ul>
          <li className={styles.arrow}>
            <Link to={`/`}>
              <IoIosArrowBack />
            </Link>
          </li>
          <li className={styles.title}>
            <Link to={`/`}>
              <h1>
                mountains forecast
              </h1>   
            </Link>
          </li>
          <li>
            <Link to={`/about`}>
              about
            </Link>
          </li>
          <li ref={ref}>
            <button onClick={() => setDropdown((prev) => !prev)}>
              models
            </button>
            {dropdown && (<ul>
              <li>
                <Link to={`/models/icon`}>
                  ICON
                </Link>
              </li>
              <li>
                <Link to={`/models/gfs`}>
                  GFS
                </Link>
              </li>
            </ul>)}
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
