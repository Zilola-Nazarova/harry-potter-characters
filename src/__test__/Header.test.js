import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('Test Header render', () => {
  test('Test Header', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
