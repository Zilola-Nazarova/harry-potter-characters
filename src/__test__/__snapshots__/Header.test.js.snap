// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header render Test Header 1`] = `
<header
  className="header"
>
  <nav
    className="navigation"
  >
    <ul>
      <li
        className="arrow"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
            />
          </svg>
        </a>
      </li>
      <li
        className="title"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          <h1>
            mountains forecast
          </h1>
        </a>
      </li>
      <li>
        <a
          className=""
          data-testid="clickAbout"
          href="/about"
          onClick={[Function]}
        >
          about
        </a>
      </li>
      <li>
        <button
          onClick={[Function]}
          type="button"
        >
          models
        </button>
      </li>
    </ul>
  </nav>
</header>
`;
